Automattic/wp-calypso
# Site Transfers: Add handling for site ownership confirmation

- Related to https://github.com/Automattic/dotcom-forge/issues/2563

## Proposed Changes

* Add a new screen to confirm the Site Transfer after clicking the link from the email.
* `/settings/start-site-transfer/${siteId}?site-transfer-confirm=${hash}`

## Testing Instructions



* Apply this diff if it's not deployed yet D112701-code
* Purchase a simple personal site
* Go to `/settings/general/${siteId}`
* Click on `Transfer site` and follow the steps.
* When you receive the email, copy the confirmation link and replace `wordpress.com` with the calypso live domain or `http://calypso.localhost:300` if you are running this branch from your computer.
* Observe a Loading bar and after a few seconds, you are redirected to `/sites`
* Access again to the already confirmed link and observe an error message.

## Screenshots

**Transferring**
<img width="1266" alt="Screenshot 2023-06-06 at 13 49 39" src="https://github.com/Automattic/wp-calypso/assets/779993/24c6c300-ee5a-4453-9554-7693c88c4682">


**Error**
<img width="1268" alt="Screenshot 2023-06-06 at 13 45 24" src="https://github.com/Automattic/wp-calypso/assets/779993/4c93d041-0fc2-4cb4-aac6-94e9ff6a1bbf">

**Success**
- The success notice was introduced to `/sites` in this PR: https://github.com/Automattic/wp-calypso/pull/77660
![image](https://github.com/Automattic/wp-calypso/assets/779993/a6b9dd48-b2a6-466a-8b17-bd03e31f7dda)

## Screencast

https://github.com/Automattic/wp-calypso/assets/779993/02afc9c2-f154-4bd0-afe4-8c7d24520970




## Pre-merge Checklist
